exonum-java-binding/core/src/main/java/com/exonum/binding/runtime/ServiceRuntime.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.runtime;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.exonum.binding.service.adapters.UserServiceAdapter;
+import com.exonum.binding.transport.Server;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import java.net.URI;
+
+/**
+ * A service runtime. It manages the services required for operation of Exonum services (e.g.,
+ * a {@link Server}; allows the native code to load and unload artifacts (JAR archives
+ * with Exonum services), create and stop services defined in the loaded artifacts.
+ *
+ * <p>This class is thread-safe and does not support client-side locking.
+ */
+final class ServiceRuntime {
+
+  private final Injector frameworkInjector;
+
+  /**
+   * Creates a new runtime with the given framework injector. Starts the server on instantiation.
+   *
+   * @param frameworkInjector the injector that has been configured with the Exonum framework 
+   *     bindings. It serves as a parent for service injectors
+   * @param serverPort a port for the web server providing transport to Java services
+   */
+  ServiceRuntime(Injector frameworkInjector, int serverPort) {
+    this.frameworkInjector = checkNotNull(frameworkInjector);
+
+    // TODO: shall we start on instantiation, or provide a separate method? Or start lazily, when
+    //    first service is started?
```

Comment:
I expect that the runtime will manage the resources required for operation of services (start/stop), hence pushed the Server#start from the bootstrap code here. WDYT?

Reply:
To answer the question we need to consider cases when the server is not needed immediately.  I don't know them.
Also, let's keep it simple, so `server.start(serverPort);` is ok here.

Reply:
Agreed, can't see any reason _not_ to start the server immediately.